examples/nidigital_spi/measurement.py

```diff
@@ -0,0 +1,184 @@
+"""Test a SPI device using an NI Digital Pattern instrument."""
+
+import contextlib
+import logging
+import pathlib
+from typing import Iterable, Tuple, Union
+
+import click
+import grpc
+import nidigital
+from _helpers import ServiceOptions
+
+import ni_measurementlink_service as nims
+
+service_directory = pathlib.Path(__file__).resolve().parent
+measurement_service = nims.MeasurementService(
+    service_config_path=service_directory / "NIDigitalSPI.serviceconfig",
+    version="0.1.0.0",
+    ui_file_paths=[service_directory / "NIDigitalSPI.measui"],
+)
+service_options = ServiceOptions()
+
+
+@measurement_service.register_measurement
+@measurement_service.configuration(
+    "pin_names",
+    nims.DataType.PinArray1D,
+    ["SPI_PINS"],
+    instrument_type=nims.session_management.INSTRUMENT_TYPE_NI_DIGITAL_PATTERN,
+)
+@measurement_service.configuration(
+    "specification_file_path", nims.DataType.Path, "Specifications.specs"
+)
+@measurement_service.configuration("levels_file_path", nims.DataType.Path, "PinLevels.digilevels")
+@measurement_service.configuration("timing_file_path", nims.DataType.Path, "Timing.digitiming")
+@measurement_service.configuration("pattern_file_path", nims.DataType.Path, "Pattern.digipat")
+# TODO: MeasurementLink UI Editor doesn't support arrays of Booleans
+@measurement_service.output("passing_sites", nims.DataType.Int32Array1D)
+@measurement_service.output("failing_sites", nims.DataType.Int32Array1D)
```

Comment:
The LabVIEW example displays an array of clusters:
![image](https://user-images.githubusercontent.com/17914399/232657857-56a3839e-4585-41f6-9c47-fccc4226e8f7.png)

My first inclination was to display parallel arrays of sites and pass/fail, similar to how the NI-DCPower Source DC Voltage example uses parallel arrays to return per-site information. However, displaying an array of Booleans is not supported yet.

Cc: @rfriedma

Reply:
This seems like the best approach for now, marking resolved.